### PR TITLE
doc: Add PendingReleaseNote for LevelDB removal

### PR DESCRIPTION
**Pull request by adamemerson**
For leveldb which was reoved in PR #43612 .

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: PendingReleaseNotes [hunk 1]
```diff
@@ -424,3 +424,6 @@
   (``auth_client_required=cephx`` and ``ms_mon_client_mode=secure``).
   If you have cephx authentication disabled on your cluster, you may
   need to adjust these settings for RGW.
+
+* LevelDB support has been removed. ``WITH_LEVELDB'' is no longer a supported
+  build option.
```
Comment (neha-ojha) [inline PendingReleaseNotes:429]: could you please add this under the `>=17.0.0` section https://github.com/ceph/ceph/blob/67d75cf37de5bf92c869c744d85a5a41fb8e7bed/PendingReleaseNotes#L1-L82

Comment (adamemerson) [reply PendingReleaseNotes]: Done, sorry. I didn't notice the other headings and thought the whole file was >=17

Comment (neha-ojha) [reply PendingReleaseNotes]: no worries, thanks!

File: PendingReleaseNotes [hunk 2]
```diff
@@ -81,6 +81,9 @@
 * A health warning will now be reported if the ``require-osd-release`` flag is not
   set to the appropriate release after a cluster upgrade.
 
+* LevelDB support has been removed. ``WITH_LEVELDB'' is no longer a supported
```
Comment (neha-ojha) [inline PendingReleaseNotes:84]: nit: use `` instead of '' after WITH_LEVELDB

**Review by neha-ojha — APPROVED** [review]

File: PendingReleaseNotes (hunk 2 above)
Comment (adamemerson) [reply PendingReleaseNotes]: Done!

**Pull request merged**